appservice/src/createAppService/AppInsightsCreateStep.ts

```diff
@@ -53,11 +55,43 @@ export class AppInsightsCreateStep extends AzureWizardExecuteStep<IAppServiceWiz
         return !wizardContext.appInsightsComponent;
     }
 
-    private async appInsightsSupportedInLocation(wizardContext: IAppServiceWizardContext, location: Location): Promise<boolean> {
+    public async appInsightsSupportedInLocation(wizardContext: IAppServiceWizardContext, location: Location): Promise<boolean> {
         const resourceClient: ResourceManagementClient = createAzureClient(wizardContext, ResourceManagementClient);
         const supportedRegions: Provider = await resourceClient.providers.get('microsoft.insights');
         const componentsResourceType: ProviderResourceType | undefined = supportedRegions.resourceTypes && supportedRegions.resourceTypes.find(aiRt => aiRt.resourceType === 'components');
 
-        return !!componentsResourceType && !!componentsResourceType.locations && componentsResourceType.locations.some((loc) => loc === location.displayName);
+        if (!!componentsResourceType && !!componentsResourceType.locations && componentsResourceType.locations.some((loc) => loc === location.displayName)) {
+            this._location = nonNullProp(location, 'name');
+            return true;
+        } else {
+            // If there is no exact match, then query the regionMapping.json
+            const request: requestUtils.Request = await requestUtils.getDefaultRequest('https://appinsights.azureedge.net/portal/regionMapping.json');
```

Comment:
If this fails, I feel like we should just return `false` (aka don't enable app insights) rather than show this error to the user. Same for if you fail to parse the response, etc.